lib/src/message/message_item.dart

```diff
@@ -139,103 +133,77 @@ class _ChatMessageItemState extends State<ChatMessageItem> with AutomaticKeepAli
   @override
   void initState() {
     super.initState();
-    _messageBloc.add(
-        RequestMessage(chatId: widget.chatId, messageId: widget.messageId, nextMessageId: widget.nextMessageId, isGroupChat: widget.isGroupChat));
+    _messageBloc
+        .add(LoadMessage(chatId: widget.chatId, messageId: widget.messageId, nextMessageId: widget.nextMessageId, isGroupChat: widget.isGroupChat));
   }
 
   @override
   bool get wantKeepAlive => true;
 
   Widget build(BuildContext context) {
     super.build(context);
-    return BlocBuilder(
-      bloc: _messageBloc,
-      builder: (context, state) {
-        if (state is MessageItemStateSuccess) {
-          _hasFile = state.hasFile;
-          _message = state.messageText;
-          _isStarred = state.isStarred;
-          String name = state.contactWrapper?.contactName;
-          String email = state.contactWrapper?.contactAddress;
-          Color color = state.contactWrapper?.contactColor;
-          return Column(
-            crossAxisAlignment: state.messageIsOutgoing ? CrossAxisAlignment.end : CrossAxisAlignment.start,
-            children: [
-              if (widget.hasDateMarker || state.showTime)
-                Padding(
-                  padding: EdgeInsets.only(bottom: messagesVerticalPadding),
-                  child: MessageDateTime(
-                    timestamp: state.messageTimestamp,
-                    hasDateMarker: widget.hasDateMarker,
-                    showTime: state.showTime,
+    return BlocProvider.value(
+      value: _messageBloc,
+      child: BlocBuilder<MessageItemBloc, MessageItemState>(
+        bloc: _messageBloc,
+        builder: (context, state) {
+          if (state is MessageItemStateSuccess) {
+            var messageStateData = state.messageStateData;
+            Widget message;
+            if (messageStateData.isInfo) {
+              message = MessageInfo(messageStateData: messageStateData, useInformationText: false);
+            } else if (messageStateData.isSetupMessage) {
+              message = MessageSetup(messageStateData: messageStateData);
+            } else if (messageStateData.isOutgoing) {
+              message = MessageSent(messageStateData: messageStateData);
+            } else {
+              message = MessageReceived(messageStateData: messageStateData);
+            }
+            return Column(
```

Comment:
That's interesting. I was expecting a row here... 🤔

_[minutes later]_
Ah, wait... It's the `MessageItem` itself. So I guess all message items has just one column which contains the items child widgets. Is that correct?

Reply:
Right, the `MessageItem` is the container for the actual message and other childs, e.g. the flag widget, date markers and similar stuff.